### ISSUE TITLE
Pay-for-order flow & wrong cart amounts

### ISSUE THREAD

**Issue by frosso** [Automattic/woocommerce-payments#8933]
### Describe the bug

When a draft (or failed) order is created in the backend, and some of its details are edited (e.g.: by removing/refunding some items), the wrong cart amount is being charged.

This is happening for all payment methods, including saved cards and PRBs.

### To Reproduce

- As a merchant, create a draft order in the backend (include billing/shipping details and some products - more than one); Save the order.
- After saving, cdit the order by removing one of the items
- As a customer go to the customer payment page
- Look at the price being charged

### Actual behavior

The incorrect price is being charged to the payment method.
PRBs show the incorrect amount & wrong currency.

### Screenshots

Video recording of the issue:
https://github.com/Automattic/woocommerce-payments/assets/273592/9d5307c9-0292-47a7-9e23-306d7459b67b

PRBs showing the incorrect amount:
<img width="1075" alt="Screenshot 2024-06-10 at 2 27 36 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/e0888f9d-8b87-4657-958b-f49ffb19e95d">
<img width="614" alt="Screenshot 2024-06-10 at 2 27 32 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/b2ca5d38-109f-465c-b318-95c953b445a0">

### Expected behavior

- The correct price should be charged
- The PRBs should show the correct price

**Comment by mdmoore** [comment]
While investigating this issue, I've been able to reproduce it in other payment gateways like WooCommerce Stripe and the core Cash on Delivery payment gateway. The issue has existed for a long time, maybe always. I tested back to WooCommerce 7.0.0 and the issue persisted. It seems like something that should be handled in core. That said, it seems like a strange use case, issuing a refund before an order has been paid. The pending order can be edited to remove items and discount prices, which will charge the correct amount.

I've reached out to the core team for input - p1725051453488789-slack-C0E1AV8T0
